client/my-sites/upgrades/index.js

```diff
@@ -284,10 +285,19 @@ module.exports = function() {
 		);
 
 		page(
-			'/checkout/:domain/:plan_name?',
+			'/checkout/:domain/:planName?',
 			adTracking.retarget,
 			controller.siteSelection,
 			upgradesController.checkout
 		);
+
+		page(
+			'/start-trial/:planName/:domain',
```

Comment:
The reason why this is highly misleading is that you can use a domain for `:site`, i.e. when a user use a custom domain as primary address. No surprise why we managed to screw that part up :).